publish.sh

```diff
@@ -68,117 +63,128 @@ tmpdir=/tmp/mediasearch
 do
-   echo preprocessing $template
-   cat cfn-templates/$template | 
-    sed -e "s%<ARTIFACT_BUCKET_TOKEN>%$BUCKET%g" | 
-    sed -e "s%<ARTIFACT_PREFIX_TOKEN>%$PREFIX%g" |
-    sed -e "s%<PYTUBELLAYER_ZIPFILE>%$pytubellayerzip%g" |
-    sed -e "s%<YTINDEXER_ZIPFILE>%$ytindexerzip%g" |
-    sed -e "s%<INDEXER_ZIPFILE>%$indexerzip%g" |
-    sed -e "s%<BUILDTRIGGER_ZIPFILE>%$buildtriggerzip%g" |
-    sed -e "s%<FINDER_ZIPFILE>%$finderzip%g" |
-    sed -e "s%<TOKEN_ENABLER_ZIPFILE>%$tokenenablerzip%g" |
-    sed -e "s%<SAMPLES_BUCKET>%$SAMPLES_BUCKET%g" |
-    sed -e "s%<SAMPLES_PREFIX>%$SAMPLES_PREFIX%g" |
-    sed -e "s%<METADATA_PREFIX>%$METADATA_PREFIX%g" |
-    sed -e "s%<OPTIONS_PREFIX>%$OPTIONS_PREFIX%g" |
-    sed -e "s%<REGION>%$region%g" > $tmpdir/$template
+  echo "Processing File " ${template}
+  cat $template | 
+  sed -e "s%<ARTIFACT_BUCKET_TOKEN>%$BUCKET%g" | 
+  sed -e "s%<ARTIFACT_PREFIX_TOKEN>%$PREFIX"/"%g" |
+  sed -e "s%<SAMPLES_BUCKET>%$SAMPLES_BUCKET%g" |
+  sed -e "s%<SAMPLES_PREFIX>%$SAMPLES_PREFIX%g" |
+  sed -e "s%<METADATA_PREFIX>%$METADATA_PREFIX%g" |
+  sed -e "s%<OPTIONS_PREFIX>%$OPTIONS_PREFIX%g" |
+  sed -e "s%<FINDER_ZIPFILE>%$finderzip%g" |
+  sed -e "s%<REGION>%$region%g" >  deploy_$template
+
+  S3PATH=s3://$BUCKET/$PREFIX/
+  aws s3 cp ${tmpdir}/${finderzip} ${S3PATH}${finderzip}
+
+  s3_template=s3://${BUCKET}/${PREFIX}/${template}
+  https_template="https://${BUCKET}.s3.${region}.amazonaws.com/${PREFIX}/${template}"
+  echo "S3 Template " $s3_template
+  echo "HTTPS Template " $https_template
+  aws cloudformation package \
+  --template-file deploy_$template \
+  --output-template-file ./out/${template} \
+  --s3-bucket $BUCKET --s3-prefix $PREFIX \
+  --region ${region} || exit 1
+  echo "Uploading template file to: ${s3_template}"
+  aws s3 cp ./out/${template} ${s3_template}
+  echo "Validating template"
+  aws cloudformation validate-template --template-url ${https_template} > /dev/null || exit 1
+  templateName=${template%.*}
+  templateNameUpper=`echo "$templateName" | awk '{print toupper($0)}'`
+  Outputs=$Outputs${templateNameUpper}" Template URL - $https_template;"
+  
+  Outputs=$Outputs${templateNameUpper}" CF Launch URL - https://${region}.console.aws.amazon.com/cloudformation/home?region=${region}#/stacks/create/review?templateURL=$https_template&stackName=MediaSearch-${templateNameUpper};"
+  rm -rf deploy_$template
 done
+popd
 
-S3PATH=s3://$BUCKET/$PREFIX
-echo "Copy $tmpdir/* to $S3PATH/"
-for f in msfinder.yaml msindexer.yaml $pytubellayerzip $ytindexerzip $indexerzip $buildtriggerzip $finderzip $tokenenablerzip
-do
-  aws s3 cp ${tmpdir}/${f} ${S3PATH}${f} || exit 1
+# Trim trailing :
+Outputs="${Outputs%;}"
+echo
+echo
+echo
+echo -e "===================="
+echo -e "      Outputs       "
+echo -e "===================="
+
+# Print strings   
+IFS=';' read -ra STRINGS <<< "$Outputs"
+for str in "${STRINGS[@]}"; do
+  echo "$str" 
+  echo
 done
 
-if $PUBLIC; then
```

Comment:
removing this code is problematic..  I had to add back equivalent code to publish the solution to our public S3 buckets.